### DISCUSSION TITLE
Use on shared-hosting without composer

### DISCUSSION THREAD

**Discussion by carliedu** [clue/framework-x#135] [Q&A]
At the homepage of framework-x.org, there is a statement that it "Runs anywhere, from shared hosting to your own servers.".
This must be better explained concerning to "shared-hosting". Usually shared hostings, does not have console access. This means, no access to composer.
So, how to run `composer require clue/framework-x:dev-main`?

**Comment by Ahmard** [comment]
Not having access to composer isn't a big deal, people usually install their composer packages locally, compress, and then upload to their hosting provider.
I think where the question lies is "How to run the server in the absence of console"

**Reply by SimonFrings** [reply]
@carliedu thanks for bringing this up, this is something I ran into a couple of times ^^

There are multiple ways to install Framework X on shared hosting. Having console access is definitely the most comfortable solution. As @Ahmard said, installing the project on you local machine and then uploading your project files is an easy way to achieve the same result.

**Reply by carliedu** [reply]
@SimonFrings I posted this discussion here on the request of @clue.
This is an problem for the shared-server administrator too, due to the opening of security breaches.
My shared-server administrator has, for a time, gave me free access to console, but every time CPanel was updated he must do it manually again. So I have give it up.
In my opinion, a tool should be created in the CPanel Control Panel (or other similar manager) to deal with composer, as this will be increasingly necessary. Today there are solution in Softaculous to thandle this need in Laravel for example (and other packages too). So the administrator do not have to do anything manually.

**Reply by clue** [reply]
What @Ahmard said ☝️ I agree that installing on your development/build platform and then transferring your files to your shared hosting provider sounds like a reasonable approach.

For what it's worth, there are also a number of shared hosting providers that also provide command line access (SSH) even in their most inexpensive tiers.

On top of this, some shared hosting control panels even integrate Composer and other PHP tools. For instance, Plesk provides a module to install PHP dependencies without command line access: https://www.plesk.com/extensions/composer/

Plenty of options in either case :+1:

**Reply by carliedu** [reply]
It exists an site called https://php-download.com/ where it is possible to download an ZIP file with autoload and vendor files.  Or prepare on an local server and zip it. But it's not the most practical.
